layers/dhcpv4.go

```diff
@@ -0,0 +1,426 @@
+// Copyright 2012 Google, Inc. All rights reserved.
+// Copyright 2009-2011 Andreas Krennmair. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file in the root of the source
+// tree.
+
+package layers
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"net"
+
+	"github.com/google/gopacket"
+)
+
+var LayerTypeDHCPv4 = gopacket.RegisterLayerType(6768, gopacket.LayerTypeMetadata{"DHCPv4", gopacket.DecodeFunc(decodeDHCPv4)})
+
+var opTypes = [3]string{
+	0: "UNSPEC",
+	1: "REQUEST",
+	2: "RESPONSE",
+}
+
+type DHCPOperation uint8
+
+const (
+	DHCP_MSG_UNSPEC DHCPOperation = iota
+	DHCP_MSG_DISCOVER
+	DHCP_MSG_OFFER
+	DHCP_MSG_REQUEST
+	DHCP_MSG_DECLINE
+	DHCP_MSG_ACK
+	DHCP_MSG_NAK
+	DHCP_MSG_RELEASE
+	DHCP_MSG_INFORM
+)
+
+var messageTypes = map[DHCPOperation]string{
+	0: "UNSPEC",
+	1: "DISCOVER",
+	2: "OFFER",
+	3: "REQUEST",
+	4: "DECLINE",
+	5: "ACK",
+	6: "NAK",
+	7: "RELEASE",
+	8: "INFORM",
+}
+
+func (o DHCPOperation) String() string {
+	return opTypes[o]
+}
+
+const (
+	_ = iota
+	DHCP_MSG_REQ
+	DHCP_MSG_RES
+)
+
+var dhcpMagic uint32 = 0x63825363
```

Comment:
For code readiness would you add here as a comment some short extract of the RFC + RFC Number, you are using to extract these magic values. That always help on the long run to know where does magic numbers come from.